### PR TITLE
[OrderByComparator] LPD 25985

### PR DESCRIPTION
**Pull request by victorhcunha**
[LPD-25985](https://liferay.atlassian.net/browse/LPD-25985)

[LPD-25985]: https://liferay.atlassian.net/browse/LPD-25985?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by victorhcunha** [comment]
ci:stop

**Comment by victorhcunha** [comment]
ci:test:sf

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:report:69282255

**Comment by victorhcunha** [comment]
ci:report:69268348

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava [hunk 1]
```diff
@@ -339,6 +341,8 @@ public class UpgradeCatchAllCheck implements ConfigurationBeanDeclaration {
 		UserLoginDateComparator.getInstance(false);
 		UserScreenNameComparator.getInstance(false);
 		WorkflowMetricsSLADefinitionVersionIdComparator.getInstance(false);
+		SavedContentEntryClassNameIdComparator.getInstance(false);
```
Comment (dantewang) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava:344]: @victorhcunha Why are these sorted at the end in this file? They should just be together with the ones starting with "S".

File: modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava [hunk 2]
```diff
@@ -454,6 +458,10 @@ public class UpgradeCatchAllCheck implements ConfigurationBeanDeclaration {
 		CommercePriceListCreateDateComparator.getInstance(showHiddenPages);
 		CommercePriceListDisplayDateComparator.getInstance(showHiddenPages);
 		CommercePriceListPriorityComparator.getInstance(showHiddenPages);
+		CommercePriceEntryUOMCreateDateComparator.getInstance(showHiddenPages);
```
Comment (dantewang) [inline modules/util/source-formatter/src/test/resources/com/liferay/source/formatter/dependencies/expected/upgrade/UpgradeCatchAllCheck.testjava:461]: @victorhcunha These 4 files are not modified in this PR. Are they missing ones from previous PRs? If so, it's better to split a separate commit and highlight it in the message.

File: modules/apps/site/site-admin-web/src/main/java/com/liferay/site/admin/web/internal/display/context/SelectSiteInitializerDisplayContext.java [hunk 3]
```diff
@@ -150,7 +150,7 @@ private List<SiteInitializerItem> _getSiteInitializerItems()
 						themeDisplay.getCompanyId(), Boolean.TRUE, null),
 					layoutSetPrototype -> new SiteInitializerItem(
 						layoutSetPrototype, themeDisplay.getLocale())),
-				new SiteInitializerNameComparator(true));
+				SiteInitializerNameComparator.getInstance(true));
```
Comment (dantewang) [inline modules/apps/site/site-admin-web/src/main/java/com/liferay/site/admin/web/internal/display/context/SelectSiteInitializerDisplayContext.java:153]: @victorhcunha This class is clearly only used as a collection comparator, so it doesn't need to be an `OrderByComparator` with singleton pattern. Please make it an ordinary `java.util.Comparator`, keep direct constructor usage.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by victorhcunha** [comment]
ci:test:sf

**Review by dantewang — APPROVED** [review]

**Comment by dantewang** [comment]
@victorhcunha If there's no unique failure and conflict, please forward, thanks!

**Comment by victorhcunha** [comment]
Thanks @dantewang !

**Comment by victorhcunha** [comment]
ci:test:upgrade

**Comment by victorhcunha** [comment]
ci:report:72574087

**Comment by victorhcunha** [comment]
ci:forward

**Comment by victorhcunha** [comment]
ci:forward

**Comment by victorhcunha** [comment]
ci:stop

**Comment by victorhcunha** [comment]
ci:forward:force

**Comment by dantewang** [comment]
ci:forward:force

**Pull request closed without merging**